src/templates/PoolV2/ShareAndEarn/ShareAndEarnCard/index.tsx

```diff
@@ -0,0 +1,183 @@
+
+  const [{ wallet, connecting }, conect] = useConnectWallet()
+  const { data } = useReferralEncrypt(wallet?.accounts[0].address)
+
+  React.useEffect(() => {
+    if (!data?.hash || parseFloat(feeJoinBroker) <= 0) return
+
+    setReferralLink(
+      `${poolLink}?tab=overview&${refferralQuery}${encodeURIComponent(
+        data.hash
+      )}`
+    )
+  }, [data])
+
+  return (
+    <S.ShareAndEarnCard>
+      <S.ShareAndEarnContainer>
+        <S.CardHeader>
+          <img
+            src="/assets/iconGradient/products-gift.svg"
+            alt=""
+            width={42}
+            height={42}
+          />
+          <S.CardTitle>Share and earn</S.CardTitle>
+        </S.CardHeader>
+
+        <S.ReferralParagraph>
+          Referral Commission {(parseFloat(feeJoinBroker) * 100).toFixed(2)}%
+        </S.ReferralParagraph>
+
+        <S.CardDescription>
+          Share your referral link with friends, family, and colleagues, and
+          earn commissions every time they sign up and make a deposit using your
+          link. It&apos;s a simple and lucrative opportunity to boost your
+          income. Start earning extra money today!
+        </S.CardDescription>
+      </S.ShareAndEarnContainer>
+
+      <S.ShareLinkContainer>
+        <S.ShareLinkContent>
+          <S.ShareLinkTitle>Share your referral link</S.ShareLinkTitle>
+
+          <S.ShareLinkWrapper>
+            <S.ShareLink
+              showCommissionUrl={
+                !!wallet && referralLink.includes(refferralQuery)
+              }
+            >
+              <span>{referralLink}</span>
+              {wallet ? (
+                <CopyToClipboard text={referralLink}>
+                  <Button
+                    className="small-button"
+                    size="medium"
+                    text="Copy"
+                    background="secondary"
+                    onClick={() => ToastInfo('link copied')}
+                  />
+                </CopyToClipboard>
+              ) : (
+                <Button
+                  className="small-button"
+                  size="medium"
+                  disabled={connecting}
+                  text="Connect Wallet"
+                  background="secondary"
+                  onClick={() => conect()}
+                />
+              )}
+            </S.ShareLink>
+            <S.ButtonWrapper>
+              {wallet ? (
+                <CopyToClipboard text={referralLink}>
+                  <Button
+                    className="medium-button"
+                    size="medium"
+                    text="Copy"
+                    onClick={() => ToastInfo('link copied')}
+                    background="secondary"
+                    fullWidth
+                  />
+                </CopyToClipboard>
+              ) : (
+                <Button
+                  className="medium-button"
+                  size="medium"
+                  onClick={() => conect()}
```

Comment:
Não manda nenhum parametro pra função, não precisa criar uma arrow func

```suggestion
                  onClick={conect}
```